### PR TITLE
[KHWEBA-84] Account dialog validation

### PR DESCRIPTION
**Pull request by gift-a**
image

### CODE REVIEW

**Review by Vladyslav-Plakhuta — DISMISSED** [review]

**Review by Vladyslav-Plakhuta — DISMISSED** [review]

**Review by OlgaOrlova — DISMISSED** [review]

File: src/app/accounts/account-dialog/account-dialog.component.ts [hunk 1]
```diff
@@ -34,16 +34,45 @@ export class AccountDialogComponent {
...
+  }
+
+  isCurrent(name) {
+    return this.data.current
+      ? name.toLowerCase() === this.data.current.name.toLowerCase()
+      : false;
+  }
+
+  exists(name) {
+    return this.data.accounts.reduce((res, acc) => {
```
Comment (ihor7016) [inline src/app/accounts/account-dialog/account-dialog.component.ts]: `accumulator` is the first parameter of the reduce callback https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce

File: src/app/accounts/account-dialog/account-dialog.component.ts [hunk 2]
```diff
@@ -34,16 +34,45 @@ export class AccountDialogComponent {
...
+
+  uniqueNameValidator(control: AbstractControl) {
+    const existing = !this.isCurrent(control.value)
+      ? this.exists(control.value)
+      : false;
+    return existing ? { existingName: { value: control.value } } : null;
+  }
+
+  isCurrent(name) {
+    return this.data.current
```
Comment (ihor7016) [inline src/app/accounts/account-dialog/account-dialog.component.ts]: this.data.current && (name.toLowerCase() === this.data.current.name.toLowerCase())

File: src/app/accounts/account-dialog/account-dialog.component.ts (hunk 1 above)
Comment (ihor7016) [inline src/app/accounts/account-dialog/account-dialog.component.ts]: you'd better use find or findIndex here

Comment (gift-a) [reply src/app/accounts/account-dialog/account-dialog.component.ts]: acc means account. Not accumulator

**Review by ihor7016 — APPROVED** [review]

**Pull request merged**